acceptance/tests/ensure_hiera_config_version.rb

```diff
@@ -0,0 +1,38 @@
+test_name 'C100304: Ensure hiera.yaml file contains version 5 format on new install' do
+
+  require 'puppet/acceptance/temp_file_utils'
+  extend Puppet::Acceptance::TempFileUtils
+
+  def environment_hiera_conf(agent)
```

Comment:
Rather than hard-coding the values for the hiera config locations in the test, I would recommend using puppet to introspect them.

The hiera config location is available as an atomic configuration value via `puppet config print hiera_config`.
The environment directory is also available via this mechanism and can be combined with the desired environment:
```
env_dir = on(agent, puppet('config', 'print', 'environmentpath'))
prod_hiera_conf = File.join ( env_dir, 'production', 'hiera.yaml')
```